src/main/java/apoc/refactor/GraphRefactoring.java

```diff
@@ -114,6 +113,91 @@
+     *
+     */
+    @Procedure(mode = Mode.WRITE)
+    @Deprecated
+    @Description("apoc.refactor.cloneSubgraph([node1,node2,...], [rel1,rel2,...], [reanchorNodePairs = [] ], [skipProperties = [] ]) YIELD input, output, error | " +
+            "clone nodes with their labels and properties (optionally skipping specific properties), and clone the given relationships (will exist between cloned nodes only or be reanchored according to reanchor node pairings)")
+    public Stream<NodeRefactorResult> cloneSubgraph(@Name("nodes") List<Node> nodes,
+                                                    @Name("rels") List<Relationship> rels,
+                                                    @Name(value = "reanchors", defaultValue = "[]") List<List<Node>> reanchors,
+                                                    @Name(value = "skipProperties", defaultValue = "[]") List<String> skipProperties) {
+        Map<Node, Node> copyMap = new LinkedHashMap<>(nodes.size());
+        Map<Node, Node> reanchorMap = reanchors.isEmpty() ? Collections.emptyMap() : new LinkedHashMap<>(reanchors.size());
+        List<NodeRefactorResult> resultStream = new ArrayList<>();
+        if (nodes == null) return Stream.empty();
+
+        // populate reanchor map
+        for (List<Node> pairing : reanchors) {
+            if (pairing == null) continue;
+
+            if (pairing.size() != 2) {
+                throw new IllegalArgumentException("\'reanchors\' must be a list of node pairs");
+            }
+
+            Node from = pairing.get(0);
+            Node to = pairing.get(1);
+
+            if (from == null || to == null) {
+                throw new IllegalArgumentException("\'reanchors\' must be a list of node pairs");
+            }
+
+            reanchorMap.put(from, to);
+        }
+
+        // clone nodes and populate copy map
+        for (Node node : nodes) {
+            if (node == null || reanchorMap.containsKey(node)) continue;
+            // nodes being reanchored from will NOT be cloned
+
+            NodeRefactorResult result = new NodeRefactorResult(node.getId());
+            try {
+                Node copy = copyLabels(node, db.createNode());
+
+                Map<String, Object> properties = node.getAllProperties();
+                if (skipProperties != null && !skipProperties.isEmpty()) {
+                    for (String skip : skipProperties) properties.remove(skip);
+                }
+                copy = copyProperties(properties, copy);
+
+                resultStream.add(result.withOther(copy));
+                copyMap.put(node, copy);
+            } catch (Exception e) {
+                resultStream.add(result.withError(e));
+            }
+        }
+
+        // clone relationships, will be between cloned nodes and/or reanchors
+        for (Relationship rel : rels) {
+            if (rel == null) continue;
+
+            Node oldStart = rel.getStartNode();
+            Node newStart = reanchorMap.get(oldStart);
```

Comment:
getOrElse

Reply:
getOrDefault()? We pay the cost of the copyMap.get() access even when not needed, but I suppose that's trivial enough for map access.